### ISSUE TITLE
Access the center of a Stencil

### ISSUE THREAD

**Issue by evetion** [rafaqz/Stencils.jl#57]
When calculating (relative) differences with a Stencil, it would be ideal to have a direct way to access the initial/center value of a Stencil, instead of having to calculate it somewhere else (probably not at compile time).

**Comment by rafaqz** [comment]
Yeah I have thought about adding a method to get it.

Do you have some code to clarify where you would like to get it?

**Comment by evetion** [comment]
Yeah, ideally one could access the center from within a custom function passed to `mapstencil`.

I'm currently using LocalFilters (pre v2), that uses three custom functions:
```julia
function TPI(dem::AbstractMatrix{<:Real}, radius = 1)
    dst = copy(dem)

    initial(a) = zero(a)  # a is the center/initial value here
    update(v, a, b) = v + a  # v is a stencil value, a the result from initial, and b the stencil mask
    # d is destination array, i the index, v the result from initial/update
    store!(d, i, v) = @inbounds d[i] = d[i] - (v - d[i]) / ((radius * 2 + 1)^2 - 1)

    return localfilter!(dst, dem, radius * 2 + 1, initial, update, store!)
end
```

Translating this to Stencils nets you
```julia
function TPI(dem::AbstractMatrix{<:Real}, window = Moore(1))
    X = mapstencil(mean, window, dem)
    dem - X
end
```

But here I have to do dem - X, while ideally I would pass something like
`x->center(x)-mean(x)`, which would save some allocations/speed.

Ideally this would also work for NamedStencils etc. And `center` might not even be the right word, because it assumes something about your stencil (which can be unbalanced), but more like `initial`/`source`/`position`, the pixel this stencil has offsets to.

**Comment by rafaqz** [comment]
That makes sense. I think `center` is ok even when its not the exact center - its the central position everything else is offset from.

The reason I haven't done this is I'm not sure reading the value can be elided in contexts where `center` is not actually called. But probably that doesn't matter.

To implement it we will need a new `center` field on all stencils and to update it when the stencil vector is updated, then the `center` will retrieve that value in your `center(x)`
